group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/ConsumerGroup.java

```diff
@@ -148,6 +151,12 @@ public static class DeadlineAndEpoch {
      */
     private final TimelineHashMap<String, TopicMetadata> subscribedTopicMetadata;
 
+    /**
+     * The consumer group's subscription model.
+     * This value is set to Homogeneous by default.
+     */
+    private final TimelineObject<ConsumerGroupSubscriptionModel> groupSubscriptionModel;
```

Comment:
nit: I think that we can drop the `group` prefix here. Based on my previous commit, we could also use `subscriptionType` here.